### PR TITLE
Addition of `pytest` case for not `None` values for `frames` and `start`/`stop`/`step`

### LINKED ISSUE #4648: Test incompatibility of `start/stop/step` and `frames` in `AnalysisBase.run(...)`
`ValueError` produced in case when both non-None `frames` and `start/stop/step` are specified, must be tested.

---

              Any reason why this isn't covered by tests? If not, then at the very least please raise an issue for it to be done later.

_Originally posted by @IAlibay in https://github.com/MDAnalysis/mdanalysis/pull/4162#discussion_r1702876788_

### PR DESCRIPTION
**Pull request by talagayev**
Fixes #4648 

Changes made in this Pull Request:
 - Added a `pytest` case in `test_base.py` to cover the case where `frames` and `start`/`stop`/`step` are both specified as not `None`, leading to an `ValueError`

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
📚 Documentation preview 📚: https://mdanalysis--4769.org.readthedocs.build/en/4769/

### CODE REVIEW

**Review by hmacdope — APPROVED** [review]
LGTM @talagayev!

File: package/CHANGELOG [hunk 1]
```diff
@@ -82,6 +82,7 @@ Enhancements
    (Issue #4673) 
  * enables parallelization for analysis.dssp.dssp.DSSP (Issue #4674)
  * Enables parallelization for analysis.hydrogenbonds.hbond_analysis.HydrogenBondAnalysis (Issue #4664)
+ * Added pytest case to cover Raise errors cases frames and step/stop/step are present (Issue #4648)
```
Comment (IAlibay) [inline package/CHANGELOG]: May I ask that we not add changelog entries for singular test additions. The changelog is really meant to be user facing "something has changed in the code". Large test changes are ok, but single test changes will clog up the changelog.

**Review by IAlibay — CHANGES_REQUESTED** [review]
Happy to discuss this further, but would prefer if we didn't fill up the changelog with small test changes.

File: package/CHANGELOG (hunk 1 above)
Comment (talagayev) [reply package/CHANGELOG]: Ah my bad, didn't know about that, but makes sense, removed the line now, should be fine now :)

**Comment by talagayev** [comment]
now that I was working on a different Issue I noticed that `test_frame_fail` in `test_base.py` possibly already covers the Issue #4648

File: package/CHANGELOG (hunk 1 above)
Comment (RMeli) [reply package/CHANGELOG]: I agree with this. Should we have a policy on what goes and what doens't go in the changelog?

**Comment by RMeli** [comment]
@talagayev can you please resolve the conflict? Thanks.

**Comment by talagayev** [comment]
@RMeli it's now good to go :)

**Review by RMeli — APPROVED** [review]
Thanks @talagayev.

Note for other reviewers: formatting will be sorted down the line, so I'm approving as-is.

**Comment by talagayev** [comment]
> Thanks @talagayev.
> 
> Note for other reviewers: formatting will be sorted down the line, so I'm approving as-is.

Happy to help :)

For the formatting for any upcoming PRs, I can run black through the added code if this would be preferred, or is
there any preference if it is formatted down the line ?

**Comment by RMeli** [comment]
> For the formatting for any upcoming PRs, I can run black through the added code if this would be preferred, or is there any preference if it is formatted down the line ?

Up to you, but in such case black should only affect the added code, not the whole file. We are using `black` too, as set up in #4725. However, we are formatting one module at a time so that we can actually check what `black` is doing.

So either way is fine. If you don't format what you are writing, it will be automatically formatted with `black` eventually.

**Comment by talagayev** [comment]
> > For the formatting for any upcoming PRs, I can run black through the added code if this would be preferred, or is there any preference if it is formatted down the line ?
> 
> Up to you, but in such case black should only affect the added code, not the whole file. We are using `black` too, as set up in #4725. However, we are formatting one module at a time so that we can actually check what `black` is doing.
> 
> So either way is fine. If you don't format what you are writing, it will be automatically formatted with `black` eventually.

Ah I see, will then run it for the code snippets that I add, although I got a feeling that sometimes `PEP8` has some small complaints for some `black` formatting

**Comment by RMeli** [comment]
Yes, this page details how to use `black` with other tools: https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html

I'll give @IAlibay a chance to review, but don't worry about the formatting.

**Comment by talagayev** [comment]
> Yes, this page details how to use `black` with other tools: https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html
> 
> I'll give @IAlibay a chance to review, but don't worry about the formatting.

Ah I see, thanks for the link, will take a look at it :)

**Review by IAlibay — APPROVED** [review]
Sorry didn't realise I was blocking here, lgtm.

**Pull request merged**